src/visitpy/visitpy/PyColorAttribute.C

```diff
@@ -74,41 +74,56 @@ ColorAttribute_SetColor(PyObject *self, PyObject *args)
 {
     ColorAttributeObject *obj = (ColorAttributeObject *)self;
 
-    unsigned char *cvals = obj->data->GetColor();
-    if(!PyArg_ParseTuple(args, "cccc", &cvals[0], &cvals[1], &cvals[2], &cvals[3]))
+    typedef unsigned char uchar;
+    PyObject *packaged_args = 0;
+    uchar *vals = obj->data->GetColor();
+
+    if (!PySequence_Check(args) || PyUnicode_Check(args))
+        return PyErr_Format(PyExc_TypeError, "Expecting a sequence of numeric args");
+
+    // break open args seq. if we think it matches this API's needs
+    if (PySequence_Size(args) == 1)
     {
-        PyObject     *tuple;
-        if(!PyArg_ParseTuple(args, "O", &tuple))
-            return NULL;
+        packaged_args = PySequence_GetItem(args, 0);
+        if (PySequence_Check(packaged_args) && !PyUnicode_Check(packaged_args) &&
+            PySequence_Size(packaged_args) == 4)
+            args = packaged_args;
+    }
 
-        if(PyTuple_Check(tuple))
+    if (PySequence_Size(args) != 4)
+    {
+        Py_XDECREF(packaged_args);
+        return PyErr_Format(PyExc_TypeError, "Expecting 4 numeric args");
+    }
+
+    for (Py_ssize_t i = 0; i < PySequence_Size(args); i++)
+    {
+        PyObject *item = PySequence_GetItem(args, i);
+
+        if (!PyNumber_Check(item))
         {
-            if(PyTuple_Size(tuple) != 4)
-                return NULL;
+            Py_DECREF(item);
+            Py_XDECREF(packaged_args);
+            return PyErr_Format(PyExc_TypeError, "arg %d is not a number type", (int) i);
+        }
+
+        long val = PyLong_AsLong(item);
+        uchar cval = uchar(val);
```

Comment:
What happens here if user passed in an integer > 255? PyLong_AsLong would succeed as far as I understand and contain a valid integer (eg NOT -1). Will the cast to uchar fail? How is this checked?

Reply:
I will test that shortly.